### ISSUE TITLE
Enable semver major action references eg: `install@v2`

### ISSUE THREAD

**Issue by jthegedus** [asdf-vm/actions#527]
As noted in the docs:

>asdf Actions do not currently support referencing by Semantic Version major only. Eg: `- uses: asdf-vm/actions/install@v2` is not supported. Work is being done to rectify this.

We should support `@v2` as most other GitHub Actions support this despite it being a terrible idea rewriting Git Tags behind peoples backs.

This is a particularly annoying practice since the whole point of asdf is to use the same tools across workflows (local dev vs CI) and developers, but if the action changes underneath people then it is technically not repeatable in the same manner... A truly bizarre decision from the GitHub Actions team for a CICD platform.

Nonetheless, the people will want it... We will recommend against it in the docs with full semver version being preferred. We will leave the note in the docs that people who care about supply chain issues should ref specific commits.

**Comment by jthegedus** [comment]
Add this to `.github/workflows/release.yaml`

```yaml
      # if there was a release, move the git tag for the major version to the same SHA as the new GitHub Release
      - name: tag major version
        if: ${{ steps.release.outputs.release_created }}
        run: |
          git config user.name github-actions[bot]
          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
          git remote add gh-token "https://${{ secrets.GITHUB_TOKEN }}@github.com/google-github-actions/release-please-action.git"
          git tag --delete v${{ steps.release.outputs.major }} || true
          git push origin :v${{ steps.release.outputs.major }} || true
          git tag --annotate v${{ steps.release.outputs.major }} --message "Release v${{ steps.release.outputs.major }}"
          git push origin v${{ steps.release.outputs.major }}

```

**Comment by jthegedus** [comment]
closed in #528
